client/state/account-recovery/reset/actions.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+} from 'state/action-types';
+
+export function fetchResetOptions( userData ) {
+	return ( dispatch ) => {
+		dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST } );
+
+		wpcom.undocumented().accountRecoveryReset( userData ).getResetOptions()
+			.then( options => dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE, options } ) )
+			.catch( error => dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR, error } ) );
+	};
+}
+
+export function fetchResetOptionsByLogin( user ) {
+	return fetchResetOptions( { user } );
+}
+
+export function fetchResetOptionsByName( firstname, lastname, url ) {
```

Comment:
> is this a sufficient representation of a user?

There is indeed a possibility for collisions here. Prior to these changes collisions might have been handled by a Happiness Engineer. For now if there is one the api can return an error.